yomigeek/mtracker-app
# #157650698 Get All User Requests API Endpoint

#### What does this PR do?
* Create an API endpoint to get all user requests

#### Description of Task to be completed?
* User should be able to view all requests logged over time through this endpoint

#### How should this be manually tested?
* Create a new project folder on your computer 
* Go to the created folder in your terminal depending on your computer operating system
* Clone the project to your local repository from git clone https://github.com/yomigeek/mtracker-app.git
* All necessary dependencies as stated in the README.md file should be installed
* run $ npm install
* Run app locally using $ npm run start
* Open the postman application and enter the URL http://localhost:5000/api/v1/users/requests with GET request selected 
* After the above is successful, you should get the following response: 
```
{
    "status": "success",
    "data": {
        "requests": [
            {
                "id": 1,
                "title": "AC repairs",
                "description": "AC power supply faulty",
                "date": "2018-17-05",
                "status": "pending",
                "priority": "high",
                "userId": 111
            },
            {
                "id": 2,
                "title": "Printer ink exhausted",
                "description": "The ink of the printer exhausted",
                "date": "2018-17-12",
                "status": "resolved",
                "priority": "medium",
                "userId": 111
            }
        ]
    }
}
```
#### Any background context you want to provide?
none

####  What are the relevant pivotal tracker stories?
story type: feature
story id: 157650698

#### What are the relevant Github Issues?
None

#### Questions:
None
